jf_agent/agent_logging.py

```diff
@@ -217,38 +246,157 @@ def filter(self, record: LogRecord) -> bool:
+                    value_repr=lambda value: _log_level_colorizer(
+                        value, LOG_LEVEL_PADDING, use_color
+                    ),
+                ),
+            ),
+            structlog.dev.Column(
+                "event",
+                structlog.dev.KeyValueColumnFormatter(
+                    key_style=None,
+                    value_style=_color_switch("event", use_color),
+                    reset_style=_color_switch("reset", use_color),
+                    value_repr=str,
+                ),
+            ),
+            # Removes the context vars from the log output
+            structlog.dev.Column(
+                "",
+                structlog.dev.KeyValueColumnFormatter(
+                    key_style=None,
+                    value_style="",
+                    reset_style="",
+                    value_repr=lambda val: "",
+                ),
+            ),
+        ],
+    )
+
+    formatter = structlog.stdlib.ProcessorFormatter(
+        foreign_pre_chain=SHARED_STRUCTLOG_PROCESSORS,
+        processors=[
+            structlog.stdlib.ProcessorFormatter.remove_processors_meta,
+            custom_console_renderer,
+        ],
+    )
+
+    return formatter
+
+
+def _log_level_colorizer(log_level: str, pad_len: int, use_color: bool) -> str:
+    log_level_padded = str(log_level + (' ' * (pad_len - len(log_level))))
+
+    if not use_color:
+        return log_level_padded
+
+    color = LOG_LEVEL_COLORS.get(log_level, colorama.Fore.WHITE)
+    return f"[{color}{log_level_padded}{colorama.Style.RESET_ALL}]"
+
+
+def _color_switch(column: str, use_color: bool) -> str:
+    if not use_color:
+        return ""
+
+    return CONSOLE_RENDERER_STYLE[column]
+
+
+def json_log_formatter() -> structlog.stdlib.ProcessorFormatter:
+    formatter = structlog.stdlib.ProcessorFormatter(
+        foreign_pre_chain=SHARED_STRUCTLOG_PROCESSORS,
+        processors=[
+            structlog.stdlib.ProcessorFormatter.remove_processors_meta,
+            structlog.processors.JSONRenderer(),
+        ],
+    )
+
+    return formatter
+
+
+def bind_default_agent_context(
+    run_mode: str,
+    company_slug: Optional[str],
+    upload_time: str,
+) -> None:
+    structlog.contextvars.clear_contextvars()
+    structlog.contextvars.bind_contextvars(
+        run_mode=run_mode,
+        company_slug=company_slug,
+        upload_time=upload_time,
+        agent_run_uuid=str(uuid.uuid4()),
+        jf_meta={'commit': os.getenv("SHA"), 'commit_build_time': os.getenv("BUILDTIME")},
+    )
+
+
 def configure(
     outdir: str, webhook_base: str, api_token: str, debug_requests=False
 ) -> AgentLoggingConfig:
+    # Remove default handlers that are added when logging is used before configuring
+    logging.getLogger().handlers.clear()
+
+    # Configure structlog before the standard logging module
+    configure_structlog()
+
+    # Set an envvar so tdqm works with structlog
+    os.environ["TDQM_POSITION"] = "-1"
```

Comment:
Nice!! cc @gavinpitt-jf might be useful elsewhere? https://tqdm.github.io/releases/#v4661-2023-08-10